src/altinn-app-frontend/src/features/form/data/submit/submitFormDataSagas.ts

```diff
@@ -116,15 +117,26 @@ function* submitComplete(state: IRuntimeState, stopWithWarnings: boolean) {
   return yield sagaPut(ProcessActions.complete());
 }
 
-export function* putFormData(state: IRuntimeState, model: any) {
+export function* putFormData({
+  state,
+  model,
+  field,
+  componentId,
+}: SaveDataParams) {
   // updates the default data element
   const defaultDataElementGuid = getCurrentTaskDataElementId(
     state.applicationMetadata.applicationMetadata,
     state.instanceData.instance,
     state.formLayout.layoutsets,
   );
   try {
-    yield call(put, dataElementUrl(defaultDataElementGuid), model);
+    const options: AxiosRequestConfig = {
+      headers: {
+        field,
+        componentId,
```

Comment:
Does this mean we're sending plain HTTP headers like `componentId: something`? I don't think that's standards compliant, as custom headers should have the `X-` prefix so we should probably send something like `X-ComponentId: something` instead.

Reply:
Ah, yes, totally agree. Fixing.